File: pset6/cash/cash.py
import cs50


def main():
    enter_cash = 0

    # Loop while user input is lower than 1
    while True:
        enter_cash = cs50.get_float("Change owed: ")

        if enter_cash > 0:
            break

    counter_change = 0

    # Round cash value to work with decimals
    enter_cash = round(enter_cash * 100)

    # Loop while input provided by user is higher or equals 25
    while enter_cash >= 25:
        # Add 1 to money change counter
        counter_change += 1
        # Subtract value 25 of cash provided by user
        enter_cash = enter_cash - 25

    # Loop while input provided by user is higher or equals 25
    while enter_cash >= 10:
        # Add 1 to money change counter
        counter_change += 1
        # Subtract value 10 of cash provided by user
        enter_cash = enter_cash - 10

    # Loop while input provided by user is higher or equals 5
    while enter_cash >= 5:
        # Add 1 to money change counter
        counter_change += 1
        # Subtract value 5 of cash provided by user
        enter_cash = enter_cash - 5

    # Loop while input provided by user is higher or equals 1
    while enter_cash >= 1:
        # Add 1 to money change counter
        counter_change += 1
        # Subtract value 1 of cash provided by user
        enter_cash = enter_cash - 1

    # Print the money change quantity
    print(counter_change)


main()

File: psets/pset1/cash/cash.c
#include <cs50.h>
#include <stdio.h>

int get_cents(void);
int calculate_quarters(int cents);
int calculate_dimes(int cents);
int calculate_nickels(int cents);
int calculate_pennies(int cents);

int main(void)
{
    // Ask how many cents the customer is owed
    int cents = get_cents();

    // Calculate the number of quarters to give the customer
    int quarters = calculate_quarters(cents);
    cents = cents - quarters * 25;

    // Calculate the number of dimes to give the customer
    int dimes = calculate_dimes(cents);
    cents = cents - dimes * 10;

    // Calculate the number of nickels to give the customer
    int nickels = calculate_nickels(cents);
    cents = cents - nickels * 5;

    // Calculate the number of pennies to give the customer
    int pennies = calculate_pennies(cents);
    cents = cents - pennies * 1;

    // Sum coins
    int coins = quarters + dimes + nickels + pennies;

    // Print total number of coins to give the customer
    printf("%i\n", coins);
}

int get_cents(void)
{
    // TODO
    int enter_cash = 0;
    // Ask user for input and loop if input is lower or equals 0
    do
    {
        enter_cash = get_int("Change owed: ");
    }
    while (enter_cash <= 0);
    return enter_cash;
}

int calculate_quarters(int cents)
{
    // TODO
    int counter = 0;

    // While cents is greater than 25 (quarters), add one to counter and
    // subtract from cents the value of a quarter
    while (cents >= 25)
    {
        counter++;
        cents = cents - 25;
    }
    return counter;
}

int calculate_dimes(int cents)
{
    // TODO
    int counter = 0;
    // While cents is greater than 10 (dimes), add one to counter and
    // subtract from cents the value of a dime
    while (cents >= 10)
    {
        counter++;
        cents = cents - 10;
    }
    return counter;
}

int calculate_nickels(int cents)
{
    // TODO
    int counter = 0;
    // While cents is greater than 5 (nickels), add one to counter and
    // subtract from cents the value of a nickel
    while (cents >= 5)
    {
        counter++;
        cents = cents - 5;
    }
    return counter;
}

int calculate_pennies(int cents)
{
    // TODO
    int counter = 0;
    // While cents is greater than 1 (pennies), add one to counter and
    // subtract from cents the value of a pennie
    while (cents >= 1)
    {
        counter++;
        cents = cents - 1;
    }
    return counter;
}
